### DISCUSSION TITLE
Add new "pH" (potential hydrogen) device class for sensors and numbers

### DISCUSSION THREAD

**Discussion by cibernox** [home-assistant/architecture#943] [Entity Models] [answered]
I wanted to propose the addition of a new class of device for sensors and numbers named "pH", as it's full name (potential hydrogen) is way less popular than it's abbreviated form.

Although most of you will know what pH is, quick chemistry class recap: pH is the measurement of how acid or alkaline a water solution is. This is a number that goes from 0 to 14, 7 being considered "neutral". pH values lower than 7 are considered acid and those higher than 7 are considered alkaline. pH is a unitless measurement. Also note that pH is capitalized with a lowercase p and an uppercase H.

The MDI icons included in home assistant has included a "pH" symbol that I've used for a while now, this icon could be assigned as the default icon for sensors of this class.

I've already went ahead and implemented it in https://github.com/home-assistant/core/pull/95928 before knowing of the existence of this repo, just to check how complicated the implementation would be (I'm fairly new to the HA codebase). It turned out to be quite simple.

**Is this feature something you would consider adding?** 

This is a value that everybody with a swimming pool or a fish tank has to monitor closely for the well being of those inside, and there are a fair amount of smart smart pool/fish-tank equipment that provide measurements for pH.

**Comment by Jc2k** [comment]
When the core team reviews this they will want to see examples of hardware you can buy in a shop that could use this if they were integrated with HA, normally this is 5 at a minimum and DIY solutions like esphome or mqtt don't count.

**Reply by cibernox** [reply]
Thanks. No problem, I’ll add a handful of examples.

**Reply by shbatm** [reply]
Here's a few I'm aware of:

- https://www.home-assistant.io/integrations/ondilo_ico/
- https://www.home-assistant.io/integrations/poolsense/
- https://www.home-assistant.io/integrations/isy994/
- https://www.home-assistant.io/integrations/omnilogic/
- https://www.home-assistant.io/integrations/flipr/

**Reply by MartinHjelmare** [reply]
With that amount of existing integrations that could use this device class, I'm in favor.

**Reply by cibernox** [reply]
If integrations in HACS count, I've developed my own which I've published in https://github.com/cibernox/homeassistant-poolstation/

This one covers the poolstation platform, which is used by a few brands of pool equipment for their smart chlorinators:
- Idegis domotic Series 2: https://www.idegis.es/en/electrolisis-salina/residencial/domotic-series-2 <-- this is the one I bought
- Astralpool Smart next: https://www.astralpool.com/en/en/water-treatment-equipment/salt-water-electrolysis/salt-water-chlorinators/smart-next.html

Other pool chlorinators that don't use the same platform:
- https://www.hendenpools.com.au/products/automation/smart-chlorinators
- https://www.bayrol.com/products/automatic-treatment/automatic-salt
- https://www.zodiac-poolcare.es/tratamiento-del-agua/cloradores-salinos/gama-exo--iq/exo--iq

Equipment for fish tanks also exist, but that's not my realm of expertise:
- https://www.fruugo.es/wifi-smart-monitor-fish-tank-aquarium-spa-pool-water-quality-meter-ph-orp-ec-ppm-fp/p-161502228-342950969
- https://www.marine-aquatics.eu/en/seneye-web-server-v2-wifi-for-seneye-systems-wwifi-module (this is the wifi module, PH/ORP probe is a separate product that work together)
- https://www.felixsmart.com/

This was just in a couple google searches, I'm sure there are docens more.

**Comment by parautenbach** [comment]
Pedantic note: As noted for the MDI icon, it's "pH" – not "PH". The title and suggestion should be renamed accordingly.

**Reply by cibernox** [reply]
Fair point, I didn't now capitalization was a big deal. I'll make sure I use the right capitalization it in the PR if this goes ahead.

**Comment by cibernox** [comment]
Friendly ping. I'm back from some holidays and if I get a 👍 on this idea, I can resume the work on https://github.com/home-assistant/core/pull/95928 and push it through the finish line rather swiftly.

**Comment by frenck** [comment, marked as answer]
We agree this one is ok to add 👍

**Comment by cibernox** [comment]
Great, I'll work on this tonight

**Comment by cibernox** [comment]
I believe the PRs for this are all ready.

**Comment by frenck** [comment]
Implemented and merged in https://github.com/home-assistant/core/pull/95928
